### PR TITLE
Vulnerability - taxtweb-popup close loop

### PR DESCRIPTION
**Pull request by nastasi-oq**
* vulnerability: fix css bug for log out dropdown menu in 'add new curve' page
  * vulnerability: improvement of taxonomy management (check on-the-fly on vulnerability and dynamic parameter for taxtweb popup).
  * vulnerabilty: a new gem taxonomy view read only field allow to see dynamically short or long version of gem taxonomy

*Vulnerabilty Type* value concurs to *Vulnerability Text* field behavior.

**NOTE**: to avoid code duplication a replacement of jquery that insist on synthetic DOM is developed so the same code can work on a real DOM (taxtweb case) or a JS structure that simulate it (vulnerability case).

### CODE REVIEW

**Comment by bwyss** [comment]
Is there a reason you are not using bootstrap?

**Comment by bwyss** [comment]
Why are we not including a QOP header/logo?

**Comment by nastasi-oq** [comment]
TaxTweb was thinked primarily as popup tool for vulnerability application (replacing Pascal TaxT standalone version) that accidentally could be used as separate web-application.
Add bootstrap and header/footer could be a new issue to be fixed further.

**Comment by daniviga** [comment]
I opened a bug (#511) about the template, I would suggest to address this issue **after** the migration to GeoNode 2.4 otherwise we will do the work twice.

File: openquakeplatform/openquakeplatform/taxtweb/static/taxtweb/js/simdom.js [hunk 1]
```diff
@@ -0,0 +1,242 @@
+function taxt_simdom()
```
Comment (daniviga) [inline openquakeplatform/openquakeplatform/taxtweb/static/taxtweb/js/simdom.js]: These empty functions are needed?

File: openquakeplatform/openquakeplatform/taxtweb/static/taxtweb/js/simdom.js [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+                if (is_set) {
+                    simDOM[name] = value;
+                }
+                else {
+                    return (simDOM[name]);
+                }
+            }
+        }
+        else {
+            simdom_log("NOT YET IMPLEMENTED 2 " + name);
```
Comment (daniviga) [inline openquakeplatform/openquakeplatform/taxtweb/static/taxtweb/js/simdom.js]: The "2" stands for?

File: openquakeplatform/openquakeplatform/taxtweb/static/taxtweb/js/simdom_post.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    noStoreysE11: null,
+    noStoreysE12: null,
+    noStoreysE21: null,
+    noStoreysE22: null,
+    noStoreysE31: null,
+    noStoreysE32: null,
+    noStoreysE41: null,
+    OccupancyCB1: null,
+    OccupancyCB2: null,
+//    OutTypeCB: taxt_OutTypeCBSelect,
```
Comment (daniviga) [inline openquakeplatform/openquakeplatform/taxtweb/static/taxtweb/js/simdom_post.js]: Really minor, use a spacing like other comments below

File: openquakeplatform/openquakeplatform/taxtweb/templates/taxtweb/checker.html [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        var ret_s = { s: "" };
+        if (populate(taxonomy, ret_s) == false) {
+            alert("Load of taxonomy failed\n" + ret_s.s);
+        }
+    }
+    $('#resultE').val(simDOM.resultE);
+    $('#id_taxonomy').on('change', run_check);
+}
+
+//-->
```
Comment (daniviga) [inline openquakeplatform/openquakeplatform/taxtweb/templates/taxtweb/checker.html]: Are you sure `//` are needed?

File: openquakeplatform/openquakeplatform/taxtweb/views.py [hunk 5]
```diff
@@ -76,17 +75,49 @@ def index(request, **kwargs):
...
+
+    if 'HTTP_HOST' in request.META:
+        proto = (request.META['HTTP_X_FORWARDED_PROTO'] if
+                 'HTTP_X_FORWARDED_PROTO' in request.META else 'http')
+        if request.META['HTTP_HOST'].startswith('taxtweb'):
+            taxt_prefix = proto + '://' + request.META['HTTP_HOST']
+        else:
+            taxt_prefix = proto + '://' + request.META['HTTP_HOST'] + '/taxtweb'
+    else:
+        taxt_prefix = "http://taxtweb.openquake.org"
```
Comment (daniviga) [inline openquakeplatform/openquakeplatform/taxtweb/views.py]: Maybe we can move this default in the settings? It's up to you

File: openquakeplatform/openquakeplatform/test/oqplatform.py [hunk 6]
```diff
@@ -237,3 +245,37 @@ def add_click_event(self):
 
     def click_at(self, x, y):
         self.driver.execute_script('pla_click(%d, %d);' % (x, y))
+
+
```
Comment (daniviga) [inline openquakeplatform/openquakeplatform/test/oqplatform.py]: Too many blank lines

File: openquakeplatform/openquakeplatform/test/oqplatform.py [hunk 7]
```diff
@@ -237,3 +245,37 @@ def add_click_event(self):
...
+    @staticmethod
+    def select_item_set(sel_obj, name):
+        for option in sel_obj.find_elements_by_tag_name('option'):
+            if option.text == name:
+                option.click() # select() in earlier versions of webdriver
```
Comment (daniviga) [inline openquakeplatform/openquakeplatform/test/oqplatform.py]: At least two spaces before inline comment

File: openquakeplatform/openquakeplatform/test/oqplatform.py [hunk 8]
```diff
@@ -237,3 +245,37 @@ def add_click_event(self):
...
+                break
+
+
+    def select_window_by_name(self, title):
```
Comment (daniviga) [inline openquakeplatform/openquakeplatform/test/oqplatform.py]: Too many blank lines

**Comment by daniviga** [comment]
- At https://github.com/gem/oq-platform/pull/507/files#diff-fac45f6ad187ddad4b2832de90706c08R24 `sys` has not been imported first
- Same file, at  L#180 a blank line is missing before `def link_has_gone_stale(): `

Please do a `pep8` run on all the `.py`

**Comment by daniviga** [comment]
A part from minor comments LGTM (changelog needs to be refreshed)

**Pull request merged**